code42_connector.py

```diff
@@ -106,7 +116,63 @@ def _handle_remove_departing_employee(self, param):
         self.client.detectionlists.departing_employee.remove(user_id)
         action_result.add_data({"userId": user_id})
         action_result.update_summary({"user_id": user_id, "username": username})
-        status_message = f"{username} was removed from the departing employee list"
+        status_message = f"{username} was removed from the departing employees list"
+        return action_result.set_status(phantom.APP_SUCCESS, status_message)
+
+    def _handle_list_departing_employees(self, param):
+        self._log_action_handler()
+        action_result = self._add_action_result(param)
+        filter_type = param.get("filter_type", DepartingEmployeeFilters.OPEN)
+        results_generator = self.client.detectionlists.departing_employee.get_all(filter_type=filter_type)
+        total_count = None
+        for page in results_generator:
+            employees = page.data.get("items", [])
+            for employee in employees:
+                action_result.add_data(employee)
+            if total_count is None:
+                total_count = page.data.get("totalCount", 0)
+
+        action_result.update_summary({"total_count": total_count})
+        status_message = "Successfully retrieved the list of departing employees"
+        return action_result.set_status(phantom.APP_SUCCESS, status_message)
+
+    def _handle_add_high_risk_employee(self, param):
+        self._log_action_handler()
+        action_result = self._add_action_result(param)
+        username = param["username"]
+        user_id = self._get_user_id(username)
+        response = self.client.detectionlists.high_risk_employee.add(user_id)
+        action_result.add_data(response.data)
+        action_result.update_summary({"user_id": user_id, "username": username})
+        status_message = f"{username} was added to the high risk employees list"
+        return action_result.set_status(phantom.APP_SUCCESS, status_message)
+
+    def _handle_remove_high_risk_employee(self, param):
+        self._log_action_handler()
+        action_result = self._add_action_result(param)
+        username = param["username"]
+        user_id = self._get_user_id(username)
+        self.client.detectionlists.high_risk_employee.remove(user_id)
+        action_result.add_data({"userId": user_id})
+        action_result.update_summary({"user_id": user_id, "username": username})
+        status_message = f"{username} was removed from the high risk employees list"
+        return action_result.set_status(phantom.APP_SUCCESS, status_message)
+
+    def _handle_list_high_risk_employees(self, param):
+        self._log_action_handler()
+        action_result = self._add_action_result(param)
+        filter_type = param.get("filter_type", HighRiskEmployeeFilters.OPEN)
+        results_generator = self.client.detectionlists.high_risk_employee.get_all(filter_type=filter_type)
+        total_count = None
+        for page in results_generator:
```

Comment:
Normally to handle `total_count` I would do something like:

```python
page = None
for page in results_generator:
    employees = page.data.get("items", [])
    for employee in employees: 
         action_result.add_data(employee)
 
total_count = page.data.get("totalCount", 0) if page else None
```
`page` will be set to the last page if there was at least one page of data.

Reply:
you could do this for departing employees too. To me this makes it more clear that we aren't acually getting the total count on each iteration.